src/numeric_index.c

```diff
@@ -336,7 +336,8 @@ IndexIterator *NewNumericRangeIterator(const IndexSpec *sp, NumericRange *nr,
                                        const NumericFilter *f) {
 
   // if this range is at either end of the filter, we need to check each record
-  if (NumericFilter_Match(f, nr->minVal) && NumericFilter_Match(f, nr->maxVal)) {
+  if (NumericFilter_Match(f, nr->minVal) && NumericFilter_Match(f, nr->maxVal) &&
+      f->geoFilter == NULL) {
```

Comment:
Why is this needed? if MIN and MAX are not matched but we do have goeFilter we still have not results no?

Reply:
If MIN/MAX are not matched it means the whole block is within range.

Reply:
I meant why you check if only if `f->geoFilter == NULL`

Reply:
This is an optimization for numeric which removes the filter in case the whole range is included. It effects `readNumeric` in `inverted_index.c`.
We want to keep this optimization for pure numeric but have to ensure that for geo we do not NULL the `NumericFilter` pointer since we need it for `isWithinRadius` function.

Reply:
Ohh I see now, nice :+1: